File: array_adt/exercises/singleMissingElement.cc
// find the single missing element in an sorted array

#include <iostream>
using namespace std;

int main(){
    int array[] = {6,7,8,9,10,12};
    int diff = array[0] - 0;
    int length = sizeof(array)/sizeof(int);

    // time complexity is O(n)
    for(int i = 0; i < length; i++){
        if (diff < (array[i] - i)){
            cout << "Missing value is " << (i+diff) << endl; // 11
        }
    }
}

File: array_adt/exercises/singleMissingElement.py
# find the single missing element in an sorted array

# input Array elements
# find difference between currentArrayValue and value's index
# repeat next 2 steps while i < arrayLength
# if difference < currentElementDifference
# then print the missing number (difference + currentIndex)

print('Enter the elements', end=' ')
array = list(map(int, input().split()))
diff = array[0] - 0

for i in range(len(array)):
    if (diff < array[i]-i):
        print(f'Missing value is {(i+diff)}')